pkg/controller/certificates/issuing/internal/secret.go

```diff
@@ -249,15 +249,25 @@ func (s *SecretsManager) getCertificateSecret(ctx context.Context, crt *cmapi.Ce
 func (s *SecretsManager) setKeystores(crt *cmapi.Certificate, secret *corev1.Secret, data SecretData) error {
 	// Handle the experimental PKCS12 support
 	if crt.Spec.Keystores != nil && crt.Spec.Keystores.PKCS12 != nil && crt.Spec.Keystores.PKCS12.Create {
+		pw := []byte{}
 		ref := crt.Spec.Keystores.PKCS12.PasswordSecretRef
-		pwSecret, err := s.secretLister.Secrets(crt.Namespace).Get(ref.Name)
-		if err != nil {
-			return fmt.Errorf("fetching PKCS12 keystore password from Secret: %v", err)
-		}
-		if pwSecret.Data == nil || len(pwSecret.Data[ref.Key]) == 0 {
-			return fmt.Errorf("PKCS12 keystore password Secret contains no data for key %q", ref.Key)
+		if ref != nil {
+			pwSecret, err := s.secretLister.Secrets(crt.Namespace).Get(ref.Name)
+			if err != nil {
+				return fmt.Errorf("fetching PKCS12 keystore password from Secret: %v", err)
+			}
+			if pwSecret.Data == nil || len(pwSecret.Data[ref.Key]) == 0 {
+				return fmt.Errorf("PKCS12 keystore password Secret contains no data for key %q", ref.Key)
+			}
+			pw = pwSecret.Data[ref.Key]
+		} else if crt.Spec.Keystores.PKCS12.Password != nil {
+			pw = []byte(*crt.Spec.Keystores.PKCS12.Password)
+		} else {
+			return fmt.Errorf("either PasswordSecretRef or Password is mandatory")
```

Comment:
I think this should be a user-facing error, ref. the comment about webhook validation above.

Reply:
+1, having a default password would make the config even simpler, and matches the idea is we shouldn't concern the user with keystore encryption given it's not relevant in kubernetes.

Reply:
it's also the default used here: https://github.com/SSLMate/go-pkcs12/blob/master/pkcs12.go#L37

Reply:
I don't think we need (or can set) a default password in this case. We should have a validating webhook to validate that either `passwordSecretRef` or `password` is set. If not, we will change the behavior if/when a user (not aware of this new feature) forgets to set `passwordSecretRef`. Cert-manager API is GA (v1) - which means we should in general not do breaking changes.

Also, having a default password on keystores is uncommon. Truststores may have a default password, as the JDK default `cacerts` has (`changeit`).

Reply:
@erikgb one could argue that supporting more inputs is not breaking behavior.
eg. not failing when no password is provided.

Reply:
Seeing how the CI is failing, I'm not clear if it's even possible to make the mandatory field PasswordSecretRef become optional (cmmeta.SecretKeySelector -> *cmmeta.SecretKeySelector) ?